### DISCUSSION TITLE
POM Version Update

### DISCUSSION THREAD

**Discussion by swaroopar** [jreleaser/jreleaser#348] [Q&A] [answered]
Hi
We are trying to use jreleaser maven plugin for a spring boot project with jenkins CI. Must say that this plugin is really making the whole release process easier. 

I'm however missing a point. Is it possible to update the version in the POMs with this plugin? One is removing the SNAPSHOT from the pom for the release. Then after release bumping the version to the next development version. 
If it is not possible, can you please suggest what is the best way to achieve it?

Thank you in advance.

**Comment by aalmiray** [comment, marked as answer]
JReleaser does not update the POM for you, you'll have to use a different plugin. There are a couple of alternatives:

- [maven-release-plugin](https://maven.apache.org/maven-release/maven-release-plugin/). You'll have to configure a `<distributionManagement>` block and invoke the `release:prepare` and `release:perform` goals.
- [maven-versions-plugin](https://www.mojohaus.org/versions-maven-plugin/set-mojo.html). Use the `versions:set` goal to set the POM(s) version to any number you want.

FWIW I personally prefer the latter option as it only updates the POM version while the former option also tags the release and performs other duties. Here's how the `versions:set` goal is used by the kcctl/kcctl project

https://github.com/kcctl/kcctl/blob/main/.github/workflows/release.yml

The workflow is split in 3 jobs:

1. set the version
2. build native executables in specific nodes
3. collect all executables and perform a release

**Reply by swaroopar** [reply]
Hi @aalmiray Thank you very much for the quick and detailed answer. Then i will also give maven-versions-plugin a try.
Thanks again for the example.
